client/lib/recommended-sites-store/store.js

```diff
@@ -42,7 +42,7 @@ function receiveRecommendations( data ) {
 		recommendations = recommendations.union( fromJS( pruned ) );
 
 		if ( recommendations !== previousRecs ) {
-			if ( recommendations.length >= MAX_RECOMMENDATIONS ) {
+			if ( recommendations.count >= MAX_RECOMMENDATIONS ) {
```

Comment:
Ah, good catch. I'll address this shortly; do you find it sensible to add
this code branch to the test suite, as it currently passes the tests?

On Tuesday, February 23, 2016, Ben Lowery <notifications@github.com> wrote:

> In client/lib/recommended-sites-store/store.js
> <https://github.com/Automattic/wp-calypso/pull/3502#discussion_r53831251>:
>
> > @@ -42,7 +42,7 @@ function receiveRecommendations( data ) {
> >  		recommendations = recommendations.union( fromJS( pruned ) );
> >
> >  		if ( recommendations !== previousRecs ) {
> > -			if ( recommendations.length >= MAX_RECOMMENDATIONS ) {
> > +			if ( recommendations.count >= MAX_RECOMMENDATIONS ) {
>
> so this is the actual problem and should be count()
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Automattic/wp-calypso/pull/3502/files#r53831251>.
>


-- 
Alexander A. Gyoshev
Web Developer / User Interface Engineer
mobile: +359 888 255838
Sofia, Bulgaria
http://gyoshev.net/